### ISSUE TITLE
fortran-linter doesn't detect OpenMP flags

### ISSUE THREAD

**Issue by fedebenelli** [cphyc/fortran-linter#17]
```
         !$OMP PARALLEL DEFAULT(NONE) &
          1
Warning: Exactly one space after comment at (1).
```

Is there a way to specify specific cases to omit? Couldn't find any in the documentation

**Comment by fedebenelli** [comment]
Also I should add that it wrongly tries to correct my `use` statements:

```
    use omp_lib
  1
Warning: Should prepend with "!$" at (1).
```

**Comment by cphyc** [comment]
Hi @fedebenelli and thanks for the report.
For the first point, could you try applying the following patch
```patch
diff --git a/fortran_linter/main.py b/fortran_linter/main.py
index 466d94a..70b7033 100755
--- a/fortran_linter/main.py
+++ b/fortran_linter/main.py
@@ -38,7 +38,7 @@ class FortranRules:
         # Fix "foo! comment" to "foo ! comment"
         (r"(\w)\!", r"\1 !", "At least one space before comment"),
         # Fix "!bar" to "! bar"
-        (r"\!(|\s\s+)(\S)", r"! \2", "Exactly one space after comment"),
+        (r"\!(|\s\s+)(?!\$(omp|OMP))(\S)", r"! \2", "Exactly one space after comment"),
         # Remove trailing ";"
         (r";\s*$", r"\n", 'Useless ";" at end of line'),
         [
```
For the second point, this is a matter of convention. Prepending the `use omp_lib` with a `!$` should in principle function when compiling with omp support _and_ without it. If you don't want this, you can always comment out the rule https://github.com/cphyc/fortran-linter/blob/767dfb441f4025b309b10301434934725e818cb4/fortran_linter/main.py#L81-L90
At the moment there is no way to deactivate specific rules apart from modifying the code, but that could easily be added and I'd be happy to guide you if you were interested in adding support for this!

**Comment by fedebenelli** [comment]
Thanks! I wasn't fully aware that `!$` worked like that, and I see that you made a pull request with the fix, thanks again!
